ecosystem/sep-0012.md

```diff
@@ -503,6 +499,72 @@ Name | Type | Description
 | Client not authenticated properly         | `401 Unauthorized` |
 | Anchor has no information on the customer | `404 Not Found`    |
 
+### Callback POST Request
+
+```
+POST [url from PUT request]
+```
+
+Name | Type | Description
+---- | ---- | -----------
+`id` | `string` | The ID of the registered customer.
+`account` | `G...` or `M...` string | Account of the regiestered customer
+`memo` | string | (optional) the client-generated [memo](https://developers.stellar.org/docs/glossary/transactions/#memo) that uniquely identifies the customer.
+`memo_type` | string | (**deprecated**, optional) type of `memo`. One of `text`, `id` or `hash`. Deprecated because memos should always be of type `id`, although anchors should continue to support this parameter for outdated clients. If `hash`, `memo` should be base64-encoded. If a memo is present in the decoded SEP-10 JWT's `sub` value, this parameter can be ignored. See the [Shared Accounts](#shared-omnibus-or-pooled-accounts) section for more information.
+`type` | `string` | (optional) the type of action the customer is being KYCd for. See the [Type Specification](#type-specification) below.
+`status` | `string` | One of the values described in [Provided Field Statuses](#provided-field-statuses)
+
+In order to validate the integrity and provenance of the request, the Anchor MUST include a signature in an additional HTTP Header `X-Stellar-Signature`.
+This Header MUST follow the specification: `X-Stellar-Signature: t=<timestamtp>, h=<host>, s=<base64 signature>` where:
+* __timestamp__ is the current timestamp at the time the callback is sent. This is used to assure the freshness of the request and to prevent this request to be replayed in the future. 
+* __host__ is the host the callback is being sent to. The host will be part of the signature. This is used to prevent malicious wallet to relay the request to another wallet
+* __base64 signature__ is the base64 encoding of the request signature. We explain below how to compute and verify this signature. The signature is computed using the Stellar private key linked to the `SIGNING_KEY` field of the anchor's [`stellar.toml`](sep-0001.md).
+
+It is the wallet's responsability to:
+* Verify the signature using the corresponding Stellar `SIGNING_KEY` field of the anchor's [`stellar.toml`](sep-0001.md).
+* Verify the freshness of the request by comparing the `timestamp` in the request with the current timestamp at the time of the reception and discard every request above a threshold of few seconds.
+* Send a working callback URL to the anchor.
+
+### VERIFY signature
+
+* Check that callback request has `X-Stellar-Signature` header
+* Parse the header and extract:
+  * Key `t`: __timestamp__
+  * Key `h`: __host__
```

Comment:
Hmm, an issue with including the host here is the receiving server is unlikely to validate it. If no one validates it, or the presence of the field here doesn't change whether the request is valid to be received by this server or not, then it won't protect against relay.

Reply:
I guess all that matters is the receiving host include their own value into their signature verification fn, and not take the value from here. That could be motivation for not including the value here since it might encourage poor practice of just using the value that is here.

Reply:
Fair point. 
We can remove the host from the header and use the `Host` HTTP header.